Add missing PINCTRL/RESET Kconfig select in several STM32 drivers

avolmat-st opened the pull request:
This PR adds missing "select RESET" in several STM32 driver which are using reset framework but miss the selection of it via Kconfig.
I noticed this while trying to build an image without CONFIG_SERIAL, leading to not having CONFIG_RESET (and thus the STM32 reset driver) built and thus missing the device-tree symbol.

Commit:
drivers: crypto: stm32: ensure RESET is selected

stm32 crypto driver requires a reset control hence ensure that
CONFIG_RESET is properly selected to allow proper build.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
drivers: display: stm32-ltdc: ensure RESET is selected

stm32 ltdc driver requires a reset control hence ensure that
CONFIG_RESET is properly selected to allow proper build.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
drivers: mipi_dsi: stm32: ensure RESET is selected

stm32 mipi dsi driver requires a reset control hence ensure that
CONFIG_RESET is properly selected to allow proper build.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
drivers: video: stm32: dcmi: ensure RESET is selected

stm32 dcmi driver requires a reset control hence ensure that
CONFIG_RESET is properly selected to allow proper build.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
drivers: video: stm32: dcmipp: ensure RESET/PINCTRL are selected

stm32 dcmipp driver requires a reset control and pinctrl
(depending on the configuration) hence ensure that
CONFIG_RESET and CONFIG_PINCTRL are properly selected to allow
proper build.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

erwango reviewed (approved):
Thanks for catching this.
It seems that some of them also miss PINCTRL. Would you mind fixing those while you're at ?

erwango reviewed (changes requested):
Cf above comment

Commit:
drivers: clock: stm32-mco: add missing select PINCTRL

stm32 mco clock driver is relying on the pinctrl framework
for configuring the MCO pin hence select CONFIG_PINCTRL
to allow proper build.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
drivers: ethernet: dwmac_stm32h7x: add select PINCTRL

stm32 dwmac_stm32h7x driver is relying on the pinctrl framework
hence select CONFIG_PINCTRL to ensure proper build.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
drivers: i3c: stm32: add select PINCTRL

stm32 i3c driver is relying on the pinctrl framework
hence select CONFIG_PINCTRL to ensure proper build.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
drivers: sdhc: stm32: add select PINCTRL

stm32 sdhc driver is relying on the pinctrl framework
hence select CONFIG_PINCTRL to ensure proper build.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
drivers: smbus: stm32: add select PINCTRL

stm32 smbus driver is relying on the pinctrl framework
hence select CONFIG_PINCTRL to ensure proper build.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
